docs/04_cv32a65x_design/source/cva6_id_stage.rst

```diff
@@ -15,15 +15,68 @@ ID_STAGE Module
 Description
 -----------
 
-The ID_STAGE module implements ... TO BE COMPLETED
+The ID_STAGE module implements the decode stage of the pipeline.
+Its main purpose is to decode RISC-V instructions coming from FRONTEND module
+(fetch stage) and send them to the ISSUE_STAGE module (issue stage).
+
+The compressed_decoder module checks whether the incoming instruction is
+compressed and output the corresponding uncompressed instruction.
+Then the decoder module decodes the instruction and send it to the
+issue stage.
+
 
 The module is connected to:
 
-* TO BE COMPLETED
+* CONTROLLER module can flush ID_STAGE decode stage
+* FRONTEND module sends instrution to ID_STAGE module
+* ISSUE module receives the decoded instruction from ID_STAGE module
+* CSR_REGFILE module sends status information about privilege mode, traps, extension support.
 
 .. include:: port_id_stage.rst
 
+
+
 Functionality
 -------------
 
 TO BE COMPLETED
+
+
+Submodules
+----------
+
+.. figure:: ../images/id_stage_modules.png
+   :name: ID_STAGE submodules
+   :align: center
+   :alt:
+
+   ID_STAGE submodules
+
+
+Compressed_decoder
+~~~~~~~~~~~~~~~~~~
+
+The compressed_decoder module decompresses all the compressed
+instructions taking a 16-bit compressed instruction and expanding it
+to its 32-bit equivalent.
+All compressed instructions have a 32-bit equivalent.
```

Comment:
This is true with CVA6 (so no change is required).
FYI: This is no more true globally for the RISC-V ISA as the Zcmp extension includes instructions like push and pop that are expanded as series of 32-bit instructions. And Zcmt instructions have no equivalent 32-bits instructions.